tests/e2e/tests/mixer/mixer_test.go

```diff
@@ -632,36 +632,57 @@ func TestNewMetrics(t *testing.T) {
 	dumpK8Env()
 	allowRuleSync()
 
-	if err := visitProductPage(productPageTimeout, http.StatusOK); err != nil {
-		fatalf(t, "Test app setup failure: %v", err)
+	var got float64
+
+	retry := util.Retrier{
+		BaseDelay: 10 * time.Second,
+		MaxDelay:  30 * time.Second,
+		Retries:   4,
 	}
 
-	log.Info("Successfully sent request(s) to /productpage; checking metrics...")
-	allowPrometheusSync()
 	promAPI, err := promAPI()
 	if err != nil {
 		fatalf(t, "Could not build prometheus API client: %v", err)
 	}
-	query := fmt.Sprintf("istio_response_bytes_count{%s=\"%s\",%s=\"200\"}", destLabel, fqdn("productpage"), responseCodeLabel)
-	t.Logf("prometheus query: %s", query)
-	value, err := promAPI.Query(context.Background(), query, time.Now())
-	if err != nil {
-		fatalf(t, "Could not get metrics from prometheus: %v", err)
+
+	retryFn := func(ctx context.Context, i int) error {
+		if err := visitProductPage(productPageTimeout, http.StatusOK); err != nil {
+			return fmt.Errorf("test app setup failure: %v", err)
+		}
+
+		t.Logf("Successfully sent request(s) to /productpage; checking metrics...")
+		allowPrometheusSync()
+		query := fmt.Sprintf("sum(istio_double_request_count{%s=\"%s\"})", "destination", "productpage-v1")
+		t.Logf("prometheus query: %s", query)
+		value, err := promAPI.Query(ctx, query, time.Now())
+		if err != nil {
+			return fmt.Errorf("could not get metrics from prometheus: %v", err)
+		}
+		t.Logf("promvalue := %s", value.String())
+
+		got, err = vectorValue(value, map[string]string{})
```

Comment:
can you use queryValue helper function here?

Reply:
Done.